packages/ra-ui-materialui/src/input/AutocompleteArrayInput.tsx

```diff
@@ -217,7 +217,7 @@ const AutocompleteArrayInput: FunctionComponent<
     // would have to first clear the input before seeing any other choices
     useEffect(() => {
         handleFilterChange('');
-    }, [...values, handleFilterChange]);
+    }, [values.length, handleFilterChange]);
```

Comment:
What if the values change but not the length? For example, by using final-form change action, we could replace a value by another and the length would not change. What do you think about the following?
```suggestion
    }, [values.join(','), handleFilterChange]);
```

Reply:
sure - that'd work